pkg/syncer/types/commontypes.go

```diff
@@ -33,12 +34,12 @@ type ConfigInfo struct {
 }
 
 var (
-	// VirtualCenter object for syncer
-	vcenter *cnsvsphere.VirtualCenter
+	// VirtualCenter instance for syncer
+	vCenterInstance *cnsvsphere.VirtualCenter
 	// Ensure vcenter is a singleton
-	onceForVirtualCenter sync.Once
-	// vcerror message from GetVirtualCenterInstance
-	vcerror error
+	vCenterInitialized uint32
```

Comment:
Since `vCenterInstance` is a pointer, you don't specifically need a variable to know if its been initialized..

Reply:
It may be the case that `vCenterInstance` is initialized during registration, but we get an error while connecting to it, in `vCenterInstance.Connect(ctx)`. We would still want to try the whole operation again.

Reply:
It may be the case that `vCenterInstance` is initialized during registration, but we might get an error while connecting to it, in `vCenterInstance.Connect`. In that case, we would want to repeat the whole operation.

`vCenterInitialized` variable is set only when both registration and connection succeed.

Reply:
Makes sense